src/overlays/actors/ovl_En_Toto/z_en_toto.c

```diff
@@ -21,86 +60,703 @@ const ActorInit En_Toto_InitVars = {
+    this->unk2B1 = this->text->unk1;
+    return 0;
+}
+
+s32 func_80BA3FCC(EnToto* this, GlobalContext* globalCtx) {
+    if (DECR(this->unk2B1) == 0) {
+        func_801518B0(globalCtx, this->text->textId, NULL);
+        return 1;
+    }
+    return 0;
+}
+
+s32 func_80BA402C(EnToto* this, GlobalContext* globalCtx) {
+    s16 prevCutscene = this->cutscene;
+
+    this->cutscene = ActorCutscene_GetAdditionalCutscene(this->cutscene);
+    ActorCutscene_SetIntentToPlay(this->cutscene);
+    ActorCutscene_Stop(prevCutscene);
+    return 0;
+}
+
+s32 func_80BA407C(EnToto* this, GlobalContext* globalCtx) {
+    if (ActorCutscene_GetCanPlayNext(this->cutscene)) {
+        ActorCutscene_StartAndSetUnkLinkFields(this->cutscene, &PLAYER->base);
+        return 1;
+    }
+    ActorCutscene_SetIntentToPlay(this->cutscene);
+    return 0;
+}
+
+s32 func_80BA40D4(EnToto* this, GlobalContext* globalCtx) {
+    if (func_80152498(&globalCtx->msgCtx) == 5 && func_80147624(globalCtx)) {
+        return 1;
+    }
+    return 0;
+}
+
+s32 func_80BA4128(EnToto* this, GlobalContext* globalCtx) {
+    if (func_80152498(&globalCtx->msgCtx) == 2) {
+        return 1;
+    }
+    return 0;
+}
+
+s32 func_80BA415C(EnToto* this, GlobalContext* globalCtx) {
+    if (func_80152498(&globalCtx->msgCtx) == 4 && func_80147624(globalCtx)) {
+        if (globalCtx->msgCtx.choiceIndex != 0) {
+            func_8019F230();
+        } else {
+            func_8019F208();
+        }
+        return (globalCtx->msgCtx.choiceIndex != 0 ? 0 : this->text->unk1) + 1; // Possible msg MACRO
+    }
+    return 0;
+}
+
+s32 func_80BA4204(EnToto* this, GlobalContext* globalCtx) {
+    EnTotoUnkStruct2* temp_v1_2;
+
+    if (DECR(this->unk2B1) == 0) {
+        if (!ENTOTO_WEEK_EVENT_FLAGS) {
+            temp_v1_2 = &D_80BA50DC[gSaveContext.playerForm - 1];
+            func_801518B0(globalCtx, (this->text->unk0 == 6 ? temp_v1_2->unk0 : temp_v1_2->unk4) & 0xFFFF, NULL);
+        }
+        return 1;
+    }
+    return 0;
+}
+
+s32 func_80BA42BC(EnToto* this, GlobalContext* globalCtx) {
+    ActorPlayer* player = PLAYER;
+    u32 phi_s0 = 0;
+    Vec3s* end = &D_80BA510C[3];
+
+    func_80BA3FB0(this, globalCtx);
+    func_800B7298(globalCtx, 0, 6);
+    if (player->base.world.pos.z > -310.0f) {
+        if ((player->base.world.pos.x > -150.0f) || (player->base.world.pos.z > -172.0f)) {
+            phi_s0 = 3;
+        } else {
+            if (player->base.world.pos.z > -232.0f) {
+                phi_s0 = 2;
+            } else {
+                phi_s0 = 1;
+            }
+        }
+    }
+    func_80122744(globalCtx, this->unk2BC, phi_s0, end - phi_s0);
+    this->unk2C4 = Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_DM_CHAR07, 0.0f, 0.0f, 0.0f, 0, 0, 0, 0xF02);
+    return 0;
+}
+
+s32 func_80BA43F4(EnToto* this, GlobalContext* globalCtx) {
+    func_80BA3C88(this);
+    if (func_80122760(globalCtx, this->unk2BC, 0x42700000)) {
```

Comment:
```suggestion
    if (func_80122760(globalCtx, this->unk2BC, 60.0f)) {
```
You'll also want to change
`s32 func_80122760(GlobalContext* globalCtx, UNK_PTR arg2, UNK_TYPE4 arg3)`
into 
`s32 func_80122760(GlobalContext* globalCtx, UNK_PTR arg1, f32 arg2)`
to make this work. The first argument is usually `arg0`, and I suppose `arg1` will also become the z_player_lib struct once that's setup.

Reply:
Agreed on this suggestion.